examples/gcf-pubsub-vm-delete-event-handler/main.py

```diff
@@ -0,0 +1,660 @@
+
+        https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry
+        """
+        return self.SEVERITY.name
+
+    def info(self):
+        """Assembles dict intended for jsonPayload"""
+        return {
+            'message': self.message(),
+            'vm_uri': self.vm_uri,
+            'result': self.result(),
+            'detail': self.detail(),
+        }
+
+    def log_entry(self):
+        """Assembles dict intended for use as LogEntry
+
+        This structure is generally passed as keyword arguments to the
+        google.cloud.logging.log_struct() function.
+        """
+        resource_labels = {
+            'function_name': self.function_name,
+            'project_id': self.function_project,
+            'region': self.function_region,
+        }
+        resource = Resource(labels=resource_labels, type='cloud_function')
+        log_entry = {
+            'log_name': self.log_name,
+            'labels': {
+                'event_id': self.event_id,
+            },
+            'severity': self.severity(),
+            'resource': resource,
+        }
+        return log_entry
+
+
+class NoOp(StructuredLog):
+    MESSAGE = "{} No action taken (NO_OP)"
+
+
+class IgnoredEventSubtype(StructuredLog):
+    """Log for when pub/sub event subtype is ignored e.g. GCE_OPERATION_DONE"""
+    MESSAGE = "No action taken, event_type is not GCE_API_CALL for {}"
+    SEVERITY = Severity.DEBUG
+    RESULT = Result.OK
+    LEVEL = logging.DEBUG
+    DETAIL = Detail.IGNORED_EVENT
+
+
+class NoMatches(StructuredLog):
+    MESSAGE = "{} matches no DNS records (NO_MATCHES)"
+    SEVERITY = Severity.DEBUG
+    RESULT = Result.OK
+    LEVEL = logging.DEBUG
+    DETAIL = Detail.NO_MATCHES
+
+
+class LostRace(StructuredLog):
+    MESSAGE = "{} does not exist, likely lost race (LOST_RACE)"
+    SEVERITY = Severity.WARNING
+    RESULT = Result.NOT_PROCESSED
+    LEVEL = logging.WARNING
+    DETAIL = Detail.LOST_RACE
+
+
+class VmNoIp(StructuredLog):
+    MESSAGE = "{} has no IP address (VM_NO_IP)"
+    SEVERITY = Severity.INFO
+    RESULT = Result.OK
+    LEVEL = logging.INFO
+    DETAIL = Detail.VM_NO_IP
+
+
+class RecordSkipped(StructuredLog):
+    MESSAGE = "{} does not match DNS record {} ({})"
+    SEVERITY = Severity.DEBUG
+    RESULT = Result.OK
+    LEVEL = logging.DEBUG
+    DETAIL = Detail.RR_MISMATCH
+
+    def __init__(self, vm_uri: str, event_id: str, record: dict):
+        super().__init__(vm_uri, event_id)
+        self.record = record
+
+    def message(self):
+        return self.MESSAGE.format(self.vm_uri, self.record['name'],
+                                   self.DETAIL.name)
+
+
+class NotARecord(RecordSkipped):
+    DETAIL = Detail.RR_NOT_A_RECORD
+
+
+class NameMismatch(RecordSkipped):
+    DETAIL = Detail.RR_NAME_MISMATCH
+
+
+class IpMismatch(RecordSkipped):
+    DETAIL = Detail.RR_IP_MISMATCH
+
+
+class RecordDeleted(StructuredLog):
+    MESSAGE = "{} matches DNS record {} deleted (RR_DELETED)"
```

Comment:
remove, see comment below. 
You're overriding `message` method so defining this class attribute and overriding the only method is just code sprawl.

Reply:
Done